hazelcast/src/main/java/com/hazelcast/internal/adapter/IMapDataStructureAdapter.java

```diff
@@ -52,8 +52,9 @@ public boolean putIfAbsent(K key, V value) {
     }
 
     @Override
+    @MethodNotAvailable
     public ICompletableFuture<Boolean> putIfAbsentAsync(K key, V value) {
-        return new SimpleCompletedFuture<Boolean>(map.putIfAbsent(key, value) == null);
+        throw new MethodNotAvailableException();
```

Comment:
This is an example of a formerly faked async method, which is now properly marked as "not available".